lib/archethic/contracts/interpreter/library.ex

```diff
@@ -217,4 +203,22 @@ defmodule Archethic.Contracts.Interpreter.Library do
       bin
     end
   end
+
+  @doc """
+  Get the genesis address of the chain
+  """
+  @spec get_genesis_address(binary()) ::
+          binary()
+  def get_genesis_address(address), do: get_first_address(address)
+
+  @doc """
+  Get the First transaction address of the transaction chain for the given address
+  """
+  @spec get_first_address(address :: binary()) ::
+          binary()
+  def get_first_address(address) do
```

Comment:
The Genesis Addresses is not a transaction or the first transaction. 
The first transaction is calulated by index = 0+1

Reply:
Indeed, genesis and first address is different. To get it, we can use the pagination of the chain.

Reply:
> Indeed, genesis and first address is different. To get it, we can use  Archethic.P2P.Message.GetFirstAddress message 

Finally, this message returns the genesis and not the first one.

Reply:
We have to see the complexity to retrieve the first transaction of a chain vs the usage. 
Is it worth it ?

Reply:
`%GetGenesisAddress{}`
`%GetFirstTxAddress{}`